static/app/components/eventsTable/eventsTable.tsx

```diff
@@ -8,26 +8,33 @@ import {Event} from 'sentry/types/event';
 type Props = {
   events: Event[];
   groupId: string;
+  orgFeatures: string[];
   orgId: string;
   projectId: string;
   tagList: Tag[];
 };
 class EventsTable extends Component<Props> {
   render() {
-    const {events, tagList, orgId, projectId, groupId} = this.props;
+    const {events, tagList, orgId, projectId, groupId, orgFeatures} = this.props;
 
     const hasUser = !!events.find(event => event.user);
 
+    const showReplayColumn = orgFeatures.includes('session-replay-ui');
+
+    const filteredTagList = tagList.filter(tag => tag.key !== 'replayId');
```

Comment:
We're pulling out the field unconditionally, and then adding it back in, as the lasts column, if we want to show it.  Seems like the wrong way around.

It would be better to only remove it if the feature is disabled... and think about writing the code so it's easy to delete the feature flag without breaking things when the time comes. 

Also, it's better to let `tagList` be in charge of the column order.

Reply:
Actually, looking at this again, there's another issue. If `replayId` is not inside `tagList` we're going to add the column anyway. `tagList` should be deciding what gets shown.

Passing `showReplayColumn` down to the `EventsTableRow` is also suspect for this reason as well.
Both here and EventsTableRow should each look at both `tagList` and `org.features` and decide what to do.